### PR TITLE
Add source connector for the PokéAPI

### PR DESCRIPTION
**Pull request by avaidyanatha**
## Main Changes
- This adds a toy source connector for the Open Source Pokémon API found here: https://pokeapi.co/

## Endpoints Implemented
- Currently only the `/pokemon` endpoint is implemented, allowing the user to retrieve one pokemon's relevant information from it.

## Notes/Self-Feedback
- We can use this as another example of building out a Python source, although it is definitely not fleshed out or meant to be used that seriously. 
- Is there a way we can differentiate between toy/tutorial connectors from the main offerings?
- No tests have been added yet

### CODE REVIEW

File: airbyte-integrations/connectors/source-pokeapi/source_pokeapi/source.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+# Basic full refresh stream
+class PokeapiStream(HttpStream):
+    url_base = 'https://pokeapi.co/api/v2/'
+
+    def __init__(self, name: str, **kwargs):
+        super().__init__()
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-pokeapi/source_pokeapi/source.py:38]: The best practice is to pass `**kwargs` to the superclass
```suggestion
        super().__init__(**kwargs)
```

File: airbyte-integrations/connectors/source-pokeapi/source_pokeapi/source.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    def check_connection(self, logger, config) -> Tuple[bool, any]:
+        input_pokemon = config['name']
+        if input_pokemon not in POKEMON_LIST:
+            return False, f"Input Pokemon {input_pokemon} is invalid. Please check your spelling our input a valid Pokemon."
+        else:
+            return True, None
+
+    def streams(self, config: Mapping[str, Any]) -> List[Stream]:
+        args = {"name": config["name"]}
+        return [Pokemon(**args)]
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-pokeapi/source_pokeapi/source.py:84]: bundling arguments into a dict like `args` is usually useful when you need to pass those params as a group. In this case we can just pass it directly
```suggestion
        return [Pokemon(name=config["name"])]
```

File: airbyte-integrations/connectors/source-pokeapi/source_pokeapi/spec.json [hunk 3]
```diff
@@ -0,0 +1,16 @@
+{
+  "documentationUrl": "https://docsurl.com",
+  "connectionSpecification": {
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "title": "Pokeapi Spec",
+    "type": "object",
+    "required": ["TODO"],
+    "additionalProperties": false,
+    "properties": {
+      "TODO: This schema defines the configuration required for the source. This usually involves metadata such as database and/or authentication information.": {
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-pokeapi/source_pokeapi/spec.json:10]: the contents of `../spec.json` should be here, and that one should be deleted.

File: airbyte-integrations/connectors/source-pokeapi/source_pokeapi/schemas/pokemon.json [hunk 4]
```diff
@@ -0,0 +1,9 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "type": "object",
+  "properties": {
+    "name": {
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-pokeapi/source_pokeapi/schemas/pokemon.json:5]: if I run `curl https://pokeapi.co/api/v2/pokemon/pikachu` the schema looks way more complicated. This should include it. This allows downstream consumers to know how to format/normalize the data.

**Review by sherifnada — CHANGES_REQUESTED** [review]
Awesome! First connector 4 days in. Good stuff.

**Comment by michel-tricot** [comment]
Told my kids about this connector, they had stars in their eyes!

File: airbyte-integrations/connectors/source-pokeapi/source_pokeapi/schemas/pokemon.json (hunk 4 above)
Comment (avaidyanatha) [reply airbyte-integrations/connectors/source-pokeapi/source_pokeapi/schemas/pokemon.json:5]: Yep makes sense, done!

File: airbyte-integrations/connectors/source-pokeapi/build.gradle [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    // For more information on standard source tests, see https://docs.airbyte.io/contributing-to-airbyte/building-new-connector/testing-connectors
+
+    // All these input paths must live inside this connector's directory (or subdirectories)
+    // TODO update the spec JSON file
+    specPath = "source_pokeapi/spec.json"
+
+    // configPath points to a config file which matches the spec.json supplied above. secrets/ is gitignored by default, so place your config file
+    // there (in case it contains any credentials)
+    // TODO update the config file to contain actual credentials
+    configPath = "secrets/config.json"
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-pokeapi/build.gradle:21]: you'll need to change this to `sample_files/config.json`

**Review by sherifnada — APPROVED** [review]
Brilliant! to publish your connector with Airbyte by default you'll need to: 

1. Add a docs page in `docs/integrations/sources/poke-api.md`
2. Add a ref to the doc page in `docs/SUMMARY.md` 
3. add an entry for this connector in `source_definitions.yaml` using a newly generated UUID 
4. create a JSON file `airbyte-config/init/src/main/resources/config/STANDARD_DESTINATION_DEFINITION` with the same info as that in #3
5. on this PR, comment `/publish connector=connectors/source-pokeapi`
6. If that is green, merge

File: airbyte-integrations/connectors/source-pokeapi/build.gradle (hunk 5 above)
Comment (avaidyanatha) [reply airbyte-integrations/connectors/source-pokeapi/build.gradle]: Missed this comment for some reason! Done.

**Pull request merged**